src/frontend/context/SecurityContext.tsx

```diff
@@ -0,0 +1,128 @@
+import * as React from "react";
+import AsyncStorage from "@react-native-async-storage/async-storage";
+import { ASYNC_KILL_KEY, PASSWORD_KEY } from "../constants";
+import { useEffect } from "react";
+import { useRef } from "react";
+
+//We should be setting the password to null when the password is turned off
+interface SecuritContextType {
+  killState: boolean;
+  setKillState: (killStateValue: boolean) => void;
+  passcode: string | null;
+  setPasscode: (passValue: string | null) => void;
+  checkFlag: boolean;
+  killStateActive: boolean;
+  setKillStateActive: (isActive: boolean) => void;
+  setCheckFlag: (checkFlag: boolean) => void;
+}
```

Comment:
wondering if there's a cleaner way of structuring this - maybe with the help of useReducer 🤔 doesn't feel right to have to pass a ton of state variables and each of their setters in a context

Reply:
Yeah, originally this was only going to have 4 values, and then it kind of blew up....Ill convert it with useReducer

Reply:
Are you thinking one reducer function to manage all states in one object? Or just one useReducer for each state?

Reply:
Ok, I cleaned it up, check out the new implementation using useReducer. It is not completed, but I have the scaffolding done. I will finish it on thursday. Thanks for the initial review, very helpful!